x/tokenfactory/client/cli/tx.go

```diff
@@ -152,3 +156,86 @@ func NewChangeAdminCmd() *cobra.Command {
 	flags.AddTxFlagsToCmd(cmd)
 	return cmd
 }
+
+func NewSetDenomMetadataCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "set-denom-metadata [metadata-file] [flags]",
+		Short: "Set metadata for a factory-created denom. Must have admin authority to do so.",
+		Long: strings.TrimSpace(
+			`Submit a parameter proposal along with an initial deposit.
+The proposal details must be supplied via a JSON file. For values that contains
```

Comment:
Is there a reason why we are referencing a proposal here?

Reply:
updated